### DISCUSSION TITLE
Skip network on mount (route pre-loading)

### DISCUSSION THREAD

**Discussion by jgoux** [urql-graphql/urql#2032] [Q&A] [answered]
Hello all,

Tanner is working on a new client-side only routing library (https://react-location.tanstack.com/) which allows for pre-loading data before transitioning to a route.

In the context of urql, I think we would need to :

1. Load the data in the route's load function using the urql client

2. Have the usual useQuery in the component rendered by the route, but have its fetch mode to "cache-only" on the first render, and switch to "cache-and-network" or "network" when the query has to run again during the component lifecycle.

It's kind of similar with the SSR workflow.

Do you have any idea how it could be done?

**Comment by JoviDeCroock** [comment, marked as answer]
You can just use the default cache-first in conjunction with the request-policy exchange and that should do the trick.

**Reply by kitten** [reply]
Also to add to this, on the topic of pre-loading data on route transitions, that very much sounds to me like a suspense + transition concern that's already possible today given a library that facilitates route transitions

**Reply by jgoux** [reply]
> You can just use the default cache-first in conjunction with the request-policy exchange and that should do the trick.

Relying on ttl in this case doesn't seem right as it's more of a "skip" matter.

Is setting the ttl to 0 enough to make the query skip the fetch and fallback to cache-and-network right away? 

When the query mode change, does it trigger a fetch?

Is it possible to opt into the request-policy exchange from the useQuery statement instead of having it centralized in the exchange config?

@kitten Yes suspense in concurrent mode with useTransition will be the way to go but right now it's not stable and I'm not aware of a router built around it yet.
Also I'm wondering if it will support lazy loaded components including data fetching (fetch the module and the data before transitioning).

**Reply by tannerlinsley** [reply]
It will support both waterfall suspense loading and parallel route-driven loading. Is all about the waterfall, as they say! The parallelism of the route loading sequence is affected by your flavor of loaders, their dependencies on parent loaders and also async imports that result in further async work to be done. Regardless how you set it up, React Location will guarantee as much parallelism as possible and ensure a single render transaction around any amount of nested routing you can come up with.
